Add new options for flexible Static Analysis rule config (#22462)
DataDog/documentation: content/en/code_analysis/static_analysis/setup.md

## Add a Static Analysis YAML file to your project

You can include the following **global** options in the `static-analysis.datadog.yml` file:

| Name               | Description                                                                                | Required | Default |
|--------------------|--------------------------------------------------------------------------------------------|----------|---------|
| `rulesets`         | A list of ruleset names and configurations. [View all available rulesets][6].              | `true`   |         |
| `ignore`           | A list of path prefixes and glob patterns to ignore. Matching files will not be analyzed.  | `false`  |         |
| `only`             | A list of path prefixes and glob patterns to analyze. Only matching files will be analyzed.| `false`  |         |
| `ignore-gitignore` | Do not use paths listed in the `.gitignore` file to skip analysis on certain files.        | `false`  | `false` |

> | `max-file-size-kb` | Ignore files larger than the specified size (unit: kB).                                    | `false`  | `200`   |

Reviewer: ```suggestion
| `max-file-size-kb` | Ignore files larger than the specified size (in kB units).                                    | `false`  | `200`   |
```

You can include the following **ruleset** options in the `static-analysis.datadog.yml` file:

| Name               | Description                                                                                                          | Required | Default |
|--------------------|----------------------------------------------------------------------------------------------------------------------|----------|---------|
| `rules`            | A list of rule configurations for rules belonging to ruleset.                                                        | `false`  |         |
| `ignore`           | A list of path prefixes and glob patterns to ignore for this specific ruleset. Matching files will not be analyzed.  | `false`  |         |

> | `only`             | A list of path prefixes and glob patterns to analyze for this specific ruleset. Only matching files will be analyzed.| `false`  |         |

Reviewer: I think you can remove the Default column here.

You can include the following **rule** options in the `static-analysis.datadog.yml` file:

| Name               | Description                                                                                                          | Required | Default |
|--------------------|----------------------------------------------------------------------------------------------------------------------|----------|---------|
| `ignore`           | A list of path prefixes and glob patterns to ignore for this specific rule. Matching files will not be analyzed.     | `false`  |         |
| `only`             | A list of path prefixes and glob patterns to analyze for this specific rule. Only matching files will be analyzed.   | `false`  |         |

> | `arguments`        | A map of values for rules that support customizable arguments.                                                       | `false`  |         |

Reviewer: I think you can remove the Default column here.

The map in the `arguments` field uses an argument's name as its key, and the values are either strings or maps:

> * If you want to set a value for the whole repository, you can specify it as a string;
> * If you want to set different values for different subtrees in the repository, you can specify them as a map from a subtree prefix to the value that the argument will have within that subtree. 

Reviewer: ```suggestion
* To set a value for the whole repository, you can specify it as a string.
* To set different values for different subtrees in the repository, you can specify them as a map from a subtree prefix to the value that the argument will have within that subtree. 
```

> Thus the full structure of the static-analysis.datadog.yml is as follows:

Reviewer: ```suggestion
The full structure of the `static-analysis.datadog.yml` file is as follows:

```

  - python-best-practices
  - python-security
  - python-code-style
    rules:
      max-function-lines:
        # Do not apply the rule max-function-lines to the following files
        ignore:
          - "src/main/util/process.py"
          - "src/main/util/datetime.py"
        arguments:

>           # Set the max-function-lines rule's threshold to 150 lines.

Reviewer: ```suggestion
          # Set the max-function-lines rule's threshold to 150 lines
```
